app/models/site.rb

```diff
@@ -4,35 +4,73 @@ class Site < ActiveRecord::Base
   
   # Redis-backed properties
   include Redis::Objects
-  hash_key :proxy_pool
+  sorted_set :proxy_pool
+  lock :proxy_pool, expiration: 60.seconds
   sorted_set :proxy_successes
   sorted_set :proxy_failures
   
+  def select_proxy(older_than=-1)
+    proxy_id, proxy_ts = nil, nil
+    # Select the least recently used proxy, get its timestamp, then update its timestamp
+    proxy_pool_lock.lock do
+      proxy_info = proxy_pool.range(0, 0, with_scores: true)
+      proxy_id, proxy_ts = proxy_info.first
+      touch(proxy_id)
+    end
+    
+    begin
```

Comment:
I chose to keep this as one unit because the first section (which has side effects in Redis) sets two values of interest to the second, which only serves to format the return value appropriately based on those two values.